src/OrbitGl/PresentEventManager.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_PRESENT_EVENT_MANAGER_H_
+#define ORBIT_GL_PRESENT_EVENT_MANAGER_H_
+
+#include <absl/synchronization/mutex.h>
+
+#include <optional>
+
+#include "ClientProtos/capture_data.pb.h"
+#include "TrackManager.h"
+#include "absl/container/flat_hash_map.h"
+
+namespace orbit_gl {
+
+// The PresentEventManager is used to categorize information about present events.
+class PresentEventManager {
```

Comment:
To be honest, I not sure why we need this class. 
The previous PR add `CaptureListener::OnPresentEvent(PresentEvent&)`. I would suggest to map `PresentEvent` to `TimerInfo` (of some new type) in the implementation of the method and pass it to `CaptureListener::OnTimer`. Therein it would be passed to `ScopeIdProvider::ProvideId`. We will have to extend its implementation, so it can handle `TimerInfo`s of the new type. 

In the end we will have a way to uniformly treat dynamically instrumented function, manual instrumentation (sync and async) and ETW. 

Or is it just a helper class for the sake of visualisation?

Reply:
This is really a minimally viable first pass that offers simple visualization through graph tracks. As discussed, @karupayun will work on creating a proper automatic frame track. I'm not sure about creating `TimerInfo`s as these are events that won't necessarily have a stop time, but it seems you have a good handle on the problem, let's talk about it offline.

Reply:
Well, a few things. I don't have too much context here, so please explain me in case I'm missing something.

First of all, I found the usage for `FrameTracks` and the new `GraphicsEtwProvider` we are creating kind of the similar thing. Both of them don't have end-time and are simply measuring distances between events to after draw something. Have both a different implementation and a different visualization for Windows and Linux doesn't have any sense IMO and I personally disagree with this as it's creating a big tech debt. Going into details, Frame Time[ms] is exactly the same and FPS is the same visualization for the same data.

Second, what I'm offering to take care is the part of the automatic Frame Track from this (https://buganizer.corp.google.com/issues/239149910). It's not a priority in my roadmap but I can take care during late Q3, but if we will have different data in both places it will mean that most work will be fixing the tech debt of using different data in both places that we should fix here. I wouldn't like to take care of this, let's try to solve it before. Sorry for being too strong about this, but the fact of introducing features with tech-debt in OrbitGl is one of the reason that later we have to do a lot of work to maintain them.

Third, I agree with both ideas, either using `TimerInfo` here as @akopich suggested or using a simplified way to save the data, but being consistent and refactoring FrameTracks to work with this data. As I told you Pierric before, we can meet and plan a solution. I believe that the inheritance between FrameTracks and TimerTracks is a big error and FrameTracks should work with just a vector of timestamps.

And lastly, I don't think we have to show two graph tracks that are basically saying the same thing. Let's be consistent and showing Frame Time as FrameTracks or either FPS but doing the same in the FrameTracks in Linux to also be drawn as GraphTracks.

Again, let me know if I'm missing something.